In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb
import scipy.stats  as st
import warnings
warnings.filterwarnings('ignore')

# task 1

h0:m<4.3
ha:m>4.3
it is left tail hypothosis 

# task 1 b calculate the critical value

In [14]:
alpha=0.1
n_c=st.norm.ppf(alpha)
n_c

-1.2815515655446004

In [3]:
alpha=.1
x=st.norm.ppf(alpha)
x

-1.2815515655446004

# task 1 c calculate hypothesis test 

In [12]:
sample=[3,4,5,5,4,5,5,4,4,4,4,4,4,4,4,5,4,4,4,3,3,3,4,3,5,4,4,5,4,4,4,5]
sample_mean=np.mean(sample)
pop_mean=4.3
pop_std=0.574
n=32

In [4]:
sample=[2,34,56,7,89,976,5454,]
samplemea=np.mean(sample)
popmean=4.3
pop_std=.567
n=32

In [7]:
z_s=(samplemea-popmean)/(pop_std/np.sqrt(n))
z_s

9389.465888191196

In [13]:
z_s=(sample_mean-pop_mean)/(pop_std/np.sqrt(n))
z_s

-2.032624022574569

# task 1 d calculate the p value

In [16]:
pvalue=st.norm.sf(abs(z_s))
pvalue

0.021045261297323523

# task 1 e conclusion

In [18]:
if z_s>n_c and pvalue>alpha:
    print('null hypothosis  is accepted')
else:
    print('null hypothosis is not accepted')

null hypothosis is not accepted


# task 2

In [24]:
cal_n=33
florida=30
cal_varience=0.028
flo_varience=0.015
alpha=0.01
it is two tail

In [8]:
data=pd.read_csv('DS1_C5_S6_Hypothesis_I_Practice_CityPrice_Data.csv')
data

FileNotFoundError: [Errno 2] No such file or directory: 'DS1_C5_S6_Hypothesis_I_Practice_CityPrice_Data.csv'

In [3]:
mean_s1=data['California'].mean()
var_s1=data['California'].var()
n1=data['California'].count()

mean_s2=data['Florida'].mean()
var_s2=data['Florida'].var()
n2=data['Florida'].count()
alpha=0.01

# task 2 b calculate the critical value 

In [23]:
Neg_Sc=st.norm.ppf(alpha/2)
Pos_Sc=st.norm.ppf(1-alpha/2)

# task 2 c calculate the hypothesis test

In [26]:
z_statistics=(mean_s1-mean_s2)/(np.sqrt(cal_varience/n1+flo_varience/n2))
z_statistics

2.7413384193834864

# task 2 d calculate the p value

In [31]:
pvalue=st.norm.sf(z_statistics)*2
pvalue

0.006118944821314794

# task 2 e conclusion

In [32]:
if Neg_Sc<z_statistics<Pos_Sc and pvalue>alpha:
    print('Accept Null Hypothesis')
else:
    print("Reject Null Hypothesis")

Reject Null Hypothesis


In [34]:
ul=(mean_s1-mean_s2)+z_statistics*(np.sqrt(cal_varience/n1+flo_varience/n2))
Ll=(mean_s1-mean_s2)-z_statistics*(np.sqrt(cal_varience/n1+flo_varience/n2))


In [35]:
ul

0.20133333333333248

In [36]:
Ll

0.0

# the 99% confidence interval to estimate the diffrence in the mean price of a liter of health drink b/w two cities is (0.20133333333333248)

# task 3

In [ ]:
h0:m<=44
ha:m>44
right tail hypothesis    

# task 3 b find the critical value

In [16]:
alpha=0.05
p_c=st.norm.ppf(1-alpha3)
p_c

1.6448536269514722

# task 3 c calculate the z stastic

In [9]:
alpha3=0.05
n1=68
pop_mean1=44
sample_mean1=45.1
pop_std1=8.7

In [14]:
z_s3=(sample_mean1-pop_mean1)/(pop_std1/np.sqrt(n1))
z_s3

1.042624411075732

# task 3 d calculate the p value

In [17]:
p_value3=st.norm.sf(abs(z_s3))
p_value3

0.14856113941662097

# task 3 e conclusion

In [18]:
if z_s3<p_c and p_value3>alpha:
    print('accept the null hypothosis')
else:
    print('reject the null hypothosis')

accept the null hypothosis
